src/ray/object_manager/tests/object_manager_test.cc

```diff
@@ -0,0 +1,152 @@
+        rpc_work_(boost::asio::make_work_guard(rpc_context_.get_executor())) {
+    ObjectManagerConfig config_;
+    config_.object_manager_address = "127.0.0.1";
+    config_.object_manager_port = 0;
+    config_.store_socket_name = "test_store_socket";
+    config_.rpc_service_threads_number = 1;
+
+    std::vector<std::thread> rpc_threads(config_.rpc_service_threads_number);
+    local_node_id_ = NodeID::FromRandom();
+    mock_gcs_client_ = std::make_unique<gcs::MockGcsClient>();
+    mock_object_directory_ = std::make_unique<MockObjectDirectory>();
+    fake_plasma_client_ = std::make_shared<plasma::FakePlasmaClient>();
+
+    object_manager_ = std::make_unique<ObjectManager>(
+        io_context_,
+        local_node_id_,
+        config_,
+        *mock_gcs_client_,
+        mock_object_directory_.get(),
+        // RestoreSpilledObjectCallback
+        [](const ObjectID &object_id,
+           int64_t object_size,
+           const std::string &object_url,
+           std::function<void(const Status &)> callback) {},
+        // get_spilled_object_url
+        [](const ObjectID &object_id) -> std::string { return ""; },
+        // pin_object
+        [](const ObjectID &object_id) -> std::unique_ptr<RayObject> { return nullptr; },
+        // fail_pull_request
+        [](const ObjectID &object_id, rpc::ErrorType error_type) {},
+        fake_plasma_client_,
+        nullptr,
+        [](const std::string &address,
+           const int port,
+           ray::rpc::ClientCallManager &client_call_manager) {
+          return std::make_shared<ray::rpc::FakeObjectManagerClient>(
+              address, port, client_call_manager);
+        },
+        rpc_context_,
+        std::move(rpc_threads));
+  }
+
+  NodeID local_node_id_;
+
+  std::unique_ptr<gcs::MockGcsClient> mock_gcs_client_;
+  std::unique_ptr<MockObjectDirectory> mock_object_directory_;
+  std::unique_ptr<ObjectManager> object_manager_;
+  std::shared_ptr<plasma::FakePlasmaClient> fake_plasma_client_;
+
+  instrumented_io_context io_context_{/*enable_lag_probe=*/false,
+                                      /*running_on_single_thread=*/true};
+  instrumented_io_context rpc_context_{/*enable_lag_probe=*/false,
+                                       /*running_on_single_thread=*/true};
+  boost::asio::executor_work_guard<boost::asio::io_context::executor_type> io_work_;
+  boost::asio::executor_work_guard<boost::asio::io_context::executor_type> rpc_work_;
+};
+
+uint32_t NumRemoteFreeObjectsRequests(const ObjectManager &object_manager) {
+  uint32_t num_free_objects_requests = 0;
+  for (const auto &[node_id, rpc_client] :
+       object_manager.remote_object_manager_clients_) {
+    auto fake_rpc_client =
+        std::dynamic_pointer_cast<ray::rpc::FakeObjectManagerClient>(rpc_client);
+    num_free_objects_requests += fake_rpc_client->num_free_objects_requests;
+  }
+  return num_free_objects_requests;
+}
+
+TEST_F(ObjectManagerTest, TestFreeObjectsIdempotent) {
```

Comment:
```suggestion
TEST_F(ObjectManagerTest, TestHandleFreeObjectsIdempotent) {
```
we're only testing idempotency of the handle, not free with local_only false, and maybe directly call HandleFree (the top level api) in the test